packages/next/client/page-loader.js

```diff
@@ -155,12 +155,13 @@ export default class PageLoader {
           return (
             (optional || param in dynamicMatches) &&
             // Interpolate group into data URL if present
-            (interpolatedRoute = interpolatedRoute.replace(
-              replaced,
-              repeat
-                ? value.map(escapePathDelimiters).join('/')
-                : escapePathDelimiters(value)
-            ))
+            (interpolatedRoute =
+              interpolatedRoute.replace(
+                replaced,
+                repeat
+                  ? value.map(escapePathDelimiters).join('/')
+                  : escapePathDelimiters(value)
+              ) || '/')
```

Comment:
@Timer Haven't investigated but it's kind of suspicious to me to have `|| '/'` here and not needing it [there](https://github.com/vercel/next.js/blob/cb96da13a0085577820855b71b7894ed607e6f1d/packages/next/build/utils.ts#L631) because most of the other logic around there is exactly the same, and I've fixed several issues before where the logic had drifted. (We should probably also look into whether it can share a bit more code)

Reply:
In this specific case, what happens is when we do `interpolatedRoute.replace('replaced', ...)`, we replace `/[[...root]]` with nothing (because it's optional) which results in an empty string, thus the necessity to do `|| '/'`.

Reply:
In the code you linked, we only replace `[[...root]]` and not `/[[...root]` (note the leading slash).